### PR TITLE
Fixes main content padding on various screens

### LINKED ISSUE #2892: Fix main content padding on various screens
**Describe the feature request.**
Update the implementation for the register, profile and reporting screens to use the padding requirements set out on this issue #2887

**Acceptance criteria**

- [x] The bottom padding for the various screens should match the updated designs listed on [this issue](https://github.com/opensrp/fhircore/issues/2887)

**Area path**
A list of ordered steps in the app on usage of the feature to support anyone testing it e.g. Code reviewer, QA e.g.
1. Login to the app
2. Open _Navigation bar_
3. Click on _Children register_
4. Click on _Child profile_
5. Click on _Edit profile_ from menu

**Implementation plan (For Engineers)**

- [x] Set bottom padding for list (LazyColumn) used to render views in profile and measure report screens, as well as the row in the register footer to `80.dp` if fab configuration exists in profile / register configuration otherwise default to `32.dp`
- [x] The UI to be updated are in `MeasureReportListScreen`, `ProfileScreen` and the `RegisterFooter` composable screens

### PR DESCRIPTION
**Pull request by brandy-kay**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #2892 

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### Demo
| Image without FAB                            | Image With FAB                                     |
| ----------------------------------- | ----------------------------------- |
| all_house_hold | edited screen |

### CODE REVIEW

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileScreen.kt [hunk 1]
```diff
@@ -154,7 +152,10 @@ fun ProfileScreen(
           color = MaterialTheme.colors.primary,
         )
       }
-      LazyColumn(state = lazyListState) {
+      LazyColumn(
+        state = lazyListState,
+        modifier = Modifier.padding(bottom = if (!fabActions.isNullOrEmpty() && fabActions.first().visible) 80.dp else 32.dp)
```
Comment (Rkareko) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileScreen.kt:157]: Convert these to constants with intuitive names e.g `FOOTER_PADDING_WITH_FAB`, `FOOTER_PADDING_WITHOUT_FAB`

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/RegisterFooter.kt [hunk 2]
```diff
@@ -51,9 +52,16 @@ fun RegisterFooter(
...
+  fabActions: List<NavigationMenuConfig>? = null,
 ) {
   if (resultCount > 0) {
-    Row(modifier = modifier.fillMaxWidth().testTag(SEARCH_FOOTER_TAG).padding(bottom = 32.dp)) {
+    Row(
+      modifier =
+        modifier
+          .fillMaxWidth()
+          .testTag(SEARCH_FOOTER_TAG)
+          .padding(bottom = if (!fabActions.isNullOrEmpty() && fabActions.first().visible) 80.dp else 32.dp)
```
[the remark this answers is not in the document]
Comment (brandy-kay) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/RegisterFooter.kt:63]: it should be a constant variable? for both

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/RegisterFooter.kt [hunk 3]
```diff
@@ -51,9 +52,19 @@ fun RegisterFooter(
...
   if (resultCount > 0) {
-    Row(modifier = modifier.fillMaxWidth().testTag(SEARCH_FOOTER_TAG).padding(bottom = 32.dp)) {
+    Row(
+      modifier =
+        modifier
+          .fillMaxWidth()
+          .testTag(SEARCH_FOOTER_TAG)
+          .padding(
+            bottom =
+              if (!fabActions.isNullOrEmpty() && fabActions.first().visible) 80.dp else 32.dp,
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/RegisterFooter.kt:65]:
```suggestion
//Declared these constants in  AndroidExtensions file
const val PADDING_BOTTOM_WITH_FAB = 80
const val PADDING_BOTTOM_WITHOUT_FAB = 32

              if (!fabActions.isNullOrEmpty() && fabActions.first().visible) PADDING_BOTTOM_WITH_FAB.dp else PADDING_BOTTOM_WITHOUT_FAB.dp,
```

**Review by ellykits — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/RegisterFooter.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ellykits) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/RegisterFooter.kt]: See my suggested code in the comments.  The idea is to define the padding values in 2 separate constants to be reused in all screens.

**Comment by ellykits** [comment]
Kindly can you share screenshots to verify this is fixed on the UI?

**Comment by brandy-kay** [comment]
@Rkareko  and @ellykits the pr is now ready for review
